src/internal/packager/helm/common.go

```diff
@@ -5,84 +5,31 @@
 package helm
 
 import (
+	"context"
 	"crypto/sha1"
 	"encoding/hex"
 	"fmt"
+	"log/slog"
 	"os"
 	"path"
 	"path/filepath"
 	"strconv"
-	"time"
 
+	"github.com/defenseunicorns/pkg/helpers/v2"
 	"github.com/zarf-dev/zarf/src/api/v1alpha1"
 	"github.com/zarf-dev/zarf/src/config"
-	"github.com/zarf-dev/zarf/src/pkg/cluster"
-	"github.com/zarf-dev/zarf/src/pkg/message"
-	"github.com/zarf-dev/zarf/src/pkg/variables"
-	"github.com/zarf-dev/zarf/src/types"
+	"github.com/zarf-dev/zarf/src/pkg/logger"
 	"helm.sh/helm/v3/pkg/action"
 	"helm.sh/helm/v3/pkg/chart"
+	"helm.sh/helm/v3/pkg/chart/loader"
+	"helm.sh/helm/v3/pkg/chartutil"
 	"helm.sh/helm/v3/pkg/cli"
+	"helm.sh/helm/v3/pkg/cli/values"
+	"helm.sh/helm/v3/pkg/getter"
 )
 
-// Helm is a config object for working with helm charts.
-type Helm struct {
-	chart      v1alpha1.ZarfChart
-	chartPath  string
-	valuesPath string
-
-	cfg     *types.PackagerConfig
-	cluster *cluster.Cluster
-	timeout time.Duration
-	retries int
-
-	kubeVersion string
-
-	chartOverride   *chart.Chart
-	valuesOverrides map[string]any
-
-	settings       *cli.EnvSettings
-	actionConfig   *action.Configuration
-	variableConfig *variables.VariableConfig
-	state          *types.ZarfState
-}
-
-// Modifier is a function that modifies the Helm config.
-type Modifier func(*Helm)
-
-// New returns a new Helm config struct.
-func New(chart v1alpha1.ZarfChart, chartPath string, valuesPath string, mods ...Modifier) *Helm {
```

Comment:
Note: Package names that match the type and constructor functions for types that have zero values that can't be used are a good practice. We don't have to bring it back for this PR but it's something to consider in the future as we continue to improve the codebase.